File: harmoniq_backend/src/main/java/dev/anirban/harmoniq_backend/constants/UrlConstants.java
package dev.anirban.harmoniq_backend.constants;

public class UrlConstants {

    // Root Endpoints for testing and running
    public static final String PUBLIC_ROUTE = "/api/root/public";
    public static final String PRIVATE_ROUTE = "/api/root/private";

    // Authentication Endpoints
    public static final String REGISTER_ENDPOINT = "/api/auth/register";
    public static final String LOGIN_ENDPOINT = "/api/auth/login";
    public static final String LOGIN_AS_GUEST_ENDPOINT = "/api/auth/login/guests";

    // User Endpoints
    public static final String USER_AVATAR_FETCH_ALL_ENDPOINT = "/api/users/avatars";
    public static final String FETCH_USER_BY_ID_ENDPOINT = "/api/users/{id}";
    public static final String UPDATE_USER_ENDPOINT = "/api/users";
    public static final String DELETE_USER_ENDPOINT = "/api/users";

    // Thread Endpoints
    public static final String CREATE_THREAD_ENDPOINT = "/api/threads";
    public static final String FETCH_THREAD_BY_ID_ENDPOINT = "/api/threads/{id}";
    public static final String FETCH_THREADS_TYPE_TAG_ENDPOINT = "/api/threads/tag/{tagName}";
    public static final String FETCH_THREADS_TYPE_PERSONALISE_ENDPOINT = "/api/threads/personalise";
    public static final String FETCH_THREADS_TYPE_POPULAR_ENDPOINT = "/api/threads/popular";
    public static final String FETCH_CURRENT_USER_THREADS_ENDPOINT = "/api/users/me/threads";
    public static final String DELETE_THREAD_BY_ID_ENDPOINT = "/api/threads/{id}";

    // Tag Endpoints
    public static final String FETCH_ALL_TAGS = "/api/tags";

    // Like Endpoints
    public static final String TOGGLE_LIKE_ENDPOINT = "/api/threads/{threadId}/likes";

    // Comment Endpoints
    public static final String CREATE_COMMENT_ENDPOINT = "/api/threads/{threadId}/comments";
    public static final String FETCH_ALL_COMMENTS_FOR_THREAD_ENDPOINT = "/api/threads/{id}/comments";

    // Conversation Endpoints
    public static final String CREATE_CONVERSATION_ENDPOINT = "/api/conversations";
    public static final String FETCH_CONVERSATION_BY_USER_ENDPOINTS = "/api/conversations";
    public static final String FETCH_CONVERSATION_HISTORY_ENDPOINT = "/api/conversations/{id}/messages";
    public static final String PROMPT_CHATBOT_ENDPOINT = "/api/conversations/{id}/messages";
    public static final String DELETE_CONVERSATION_ENDPOINT = "/api/conversations/{id}";
}

File: harmoniq_frontend/src/app/shared/constants/url-constants.ts
import { environment } from 'src/environments/environment';

// Base Url
export const BASE_URL = `${environment.apiBaseUrl}/api`;

// Chatbot Avatar URl
export const CHATBOT_AVATAR_URL = `${environment.apiBaseUrl}/avatars/chatbot/chatbot.jpg`;

// Authentication Endpoints
export const REGISTER_ENDPOINT = `${BASE_URL}/auth/register`;
export const LOGIN_ENDPOINT = `${BASE_URL}/auth/login`;
export const LOGIN_AS_GUEST = `${BASE_URL}/auth/login/guests`;

// User Endpoints
export const USER_AVATAR_FETCH_ALL_ENDPOINT = `${BASE_URL}/users/avatars`;
export const FETCH_USER_BY_ID_ENDPOINT = `${BASE_URL}/users/:id`;
export const UPDATE_USER_ENDPOINT = `${BASE_URL}/users`;
export const DELETE_USER_ENDPOINT = `${BASE_URL}/users`;

// Threads Endpoints
export const CREATE_THREAD_ENDPOINT = `${BASE_URL}/threads`;
export const FETCH_THREAD_BY_ID_ENDPOINT = `${BASE_URL}/threads/:id`;
export const FETCH_THREAD_TYPE_TAG_ENDPOINT = `${BASE_URL}/threads/tag/:tagName?page=:page&size=:size`;
export const FETCH_THREAD_TYPE_PERSONALISED_ENDPOINT = `${BASE_URL}/threads/personalise?page=:page&size=:size`;
export const FETCH_THREAD_TYPE_POPULAR_ENDPOINT = `${BASE_URL}/threads/popular?page=:page&size=:size`;
export const FETCH_CURRENT_USER_THREADS_ENDPOINT = `${BASE_URL}/users/me/threads?page=:page&size=:size`;
export const DELETE_THREAD_BY_ID_ENDPOINT = `${BASE_URL}/threads/:id`;

// Tag Endpoints
export const FETCH_ALL_TAGS_ENDPOINT = `${BASE_URL}/tags`;

// Like Endpoints
export const TOGGLE_LIKE_ENDPOINTS = `${BASE_URL}/threads/:id/likes`;

// Comment Endpoints
export const CREATE_COMMENT_ENDPOINT = `${BASE_URL}/threads/:threadId/comments`;
export const FETCH_ALL_COMMENTS_FOR_THREAD_ENDPOINT = `${BASE_URL}/threads/:threadId/comments?page=:page&size=:size`;

// Conversation Endpoints
export const CREATE_CONVERSATION_ENDPOINT = `${BASE_URL}/conversations`;
export const FETCH_USER_CONVERSATIONS = `${BASE_URL}/conversations?page=:page&size=:size`;
export const SEND_MESSAGE = `${BASE_URL}/conversations/:id/messages`;
export const FETCH_CONVERSATION_HISTORY = `${BASE_URL}/conversations/:id/messages?page=:page&size=:size`;
export const DELETE_CONVERSATION_ENDPOINT = `${BASE_URL}/conversations/:id`;
